### ISSUE TITLE
Mocking Behavior Similar To Rspec: expect to receive and return 

### ISSUE THREAD

**Issue by robskrob** [testdouble/testdouble.js#413]
### Description
I'm trying to assert that a JS object has received a method invocation and also returns a value, similar to this [rspec functionality](https://github.com/rspec/rspec-mocks/blob/master/lib/rspec/mocks/example_methods.rb#L302). I have read [your documentation](https://github.com/testdouble/testdouble.js/blob/master/docs/6-verifying-invocations.md), but I am still at a lost as to how to accomplish my goal of writing something like the following

### Example
Here's an example of what I am trying to do:
```js
user = td.object(new User());
td.verify(user.create(12)).thenResolve({some: "object"});

signUp = new SignUp(params, user);

signUp.start();
```

Do you have any examples that show how to demonstrate the above assertion?

**Comment by searls** [comment]
**tl;dr** use `td.when`  instead of `td.verify` and then add an assertion based on what your subject is actually meant to do.

What you're describing is a stub (`thenResolve`) + mock (pre-emptive expectation), which testdouble intentionally does not support. Anything that must be stubbed in order for a test to work necessarily doesn't need to be verified. That is to say: if the stubbing is truly necessary, the desired behavior would not happen if the call didn't take place, so explicit [verification would be redundant](https://github.com/testdouble/testdouble.js/blob/master/docs/B-frequently-asked-questions.md#why-shouldnt-i-call-both-tdwhen-and-tdverify-for-a-single-interaction-with-a-test-double).

Additionally, if the return value of the call matters, then asserting the call itself took place is insufficient, and some other behavior should be measured and asserted on. I don't know what that is, but perhaps `start()` either returns something (or should return something, if practicing TDD and using the test as a sounding board for your design)  or has some other side effect that can be measured.

```js
user = td.object(new User());
td.when(user.create(12)).thenResolve({some: "object"});

signUp = new SignUp(params, user);

signUp.start();

// Assert based on what happens after `{some: "object"}` is resolved
```

This tends to be confusing for people at first. If you provide a code sample I can provide an example. As for RSpec, I'd strongly recommend anyone avoid adopting its mocking library.

**Comment by robskrob** [comment]
@searls thank you for this detailed reply.
